core/src/main/java/com/graphhopper/routing/GraphHopperRouter.java

```diff
@@ -0,0 +1,322 @@
+import com.graphhopper.util.exceptions.PointOutOfBoundsException;
+import com.graphhopper.util.shapes.BBox;
+import com.graphhopper.util.shapes.GHPoint;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.graphhopper.routing.weighting.TurnCostProvider.NO_TURN_COST_PROVIDER;
+import static com.graphhopper.routing.weighting.Weighting.INFINITE_U_TURN_COSTS;
+import static com.graphhopper.util.Helper.DIST_3D;
+import static com.graphhopper.util.Parameters.Algorithms.*;
+import static com.graphhopper.util.Parameters.Routing.CURBSIDE;
+
+public class GraphHopperRouter {
+    private final EncodingManager encodingManager;
+    private final GraphHopperStorage ghStorage;
+    private final LocationIndex locationIndex;
+    private final LMPreparationHandler lmPreparationHandler;
+    private final List<CHProfile> chProfiles;
+    // todonow: too costly to import all the time, do static init or pass via constructor?
+    private TranslationMap trMap = new TranslationMap().doImport();
+    private RoutingConfig routingConfig;
+    private WeightingFactory weightingFactory = new DefaultWeightingFactory();
+    private PathDetailsBuilderFactory pathBuilderFactory = new PathDetailsBuilderFactory();
+    private boolean chDisablingAllowed;
+    private boolean lmDisablingAllowed;
+    private boolean chEnabled;
+    private boolean lmEnabled;
+
+    public GraphHopperRouter(GraphHopperStorage ghStorage, LocationIndex locationIndex,
+                             // todonow: remove these dependencies
+                             LMPreparationHandler lmPreparationHandler, CHPreparationHandler chPreparationHandler,
+                             RoutingConfig routingConfig) {
+        if (ghStorage.isClosed())
+            throw new IllegalStateException("GH storage should not be closed");
+
+        this.ghStorage = ghStorage;
+        this.chProfiles = ghStorage.getCHProfiles();
+        this.encodingManager = ghStorage.getEncodingManager();
+        this.locationIndex = locationIndex;
+        this.lmPreparationHandler = lmPreparationHandler;
+        this.routingConfig = routingConfig;
+        // todonow: maybe simply add setters for ch/lm enabled -> no need for lm/ch prep handlers here anymore
+        chEnabled = chPreparationHandler.isEnabled();
+        lmEnabled = lmPreparationHandler.isEnabled();
+        // todonow: maybe these could go into routing config -> no need for lm/ch prep handlers here anymore
+        chDisablingAllowed = chPreparationHandler.isDisablingAllowed();
+        lmDisablingAllowed = lmPreparationHandler.isDisablingAllowed();
+    }
+
+    // todonow: these setters are not thread-safe, do we care? maybe use builder pattern instead? or just volatile?
+    // at least document thread safety of this class, also there is lock mechanism in GraphHopper but what if this class
+    // is used standalone?
```

Comment:
Maybe we get rid of the change graph endpoint and its lock mechnism for now to reduce complexity?

My plan was also for #1280 to reduce the number of possibilities to modify the base graph and it seems that spatial rules can replace change graph endpoint in combination with CustomWeighting.

Reply:
Ser getting rid of this lock mechanism would certainly make things a bit simpler...

Reply:
Yes, the change graph endpoint was supposed to do only smaller changes and increases the weight only so that it would work with LM at least, but yeah, it is not really battle tested and probably also not used.

Reply:
So we remove it for 1.0?

Reply:
Ok, #1931